exercises/concept/treasure-factory/tests/Tests.elm

```diff
@@ -0,0 +1,217 @@
+module Tests exposing (tests)
+
+import Expect
+import Test exposing (Test, describe, test)
+import TreasureFactory
+
+
+tests : Test
+tests =
+    describe "TreasureFactory"
+        [ describe "1"
+            [ test "module compiles" <|
+                \() -> Expect.pass
+            , test "makeChest returns a value" <|
+                \() ->
+                    TreasureFactory.makeChest "password" 1
+                        |> Expect.equal (TreasureFactory.makeChest "password" 1)
+            , test "makeChest returns a value for different types of treasure" <|
+                \() ->
+                    TreasureFactory.makeChest "password" "gold"
+                        |> Expect.equal (TreasureFactory.makeChest "password" "gold")
+            , test "makeChest doesn't internally use a fixed password" <|
+                \() ->
+                    TreasureFactory.makeChest "password" 1
+                        /= TreasureFactory.makeChest "p4$$w0rd" 1
+                        |> Expect.true "Chests created with different passwords should not be the same"
+            , test "makeChest doesn't internally use a fixed treasure" <|
+                \() ->
+                    TreasureFactory.makeChest "password" 1
+                        /= TreasureFactory.makeChest "password" 2
+                        |> Expect.true "Chests created with different treasures should not be the same"
+            ]
+        , describe "2"
+            [ test "chest with short passwords are insecure" <|
+                \() ->
+                    TreasureFactory.secureChest (TreasureFactory.makeChest "12345" 1)
+                        |> Expect.equal Nothing
+            , test "chest with passwords of 11 characters are insecure" <|
+                \() ->
+                    TreasureFactory.secureChest (TreasureFactory.makeChest "12345678901" 1)
+                        |> Expect.equal Nothing
+            , test "chest with passwords of 12 characters are secure" <|
+                \() ->
+                    TreasureFactory.secureChest (TreasureFactory.makeChest "123456789012" 1)
+                        /= Nothing
+                        |> Expect.true "Chests with 12 characters should be secure"
+            , test "chest with passwords of more than 12 characters are secure" <|
+                \() ->
+                    TreasureFactory.secureChest (TreasureFactory.makeChest "123456789012345789" 1)
+                        /= Nothing
+                        |> Expect.true "Chests with 12 characters should be secure"
+            ]
+        , describe "3"
+            [ test "chests with non-unique treasures are removed" <|
```

Comment:
OK, I will remove the tests that check for password in task 3, they are not part of the requirements.
I will also remove the tests that check for length since they are redundant with the ones checking for values.